### PR TITLE
Replace fastSim.isChosen() with toModify/toReplaceWith/makeProcessModifier

### PR DESCRIPTION
**Pull request by makortel**
This PR is an attempt to get rid of the (disallowed) `Modifier.isChosen()` for FastSim customizations. Mostly it was a matter of using `toModify()` or `toReplaceWith()`, but the "digi aliases" required `ProcessModifier` plus two hacks to be able to track whether the job is classical mixing, premixing step1, or premixing step2 (from "which `_cff` files have been loaded").

I also catched two typos in the FastSim customization in `Validation/RecoEgamma/python/photonPostProcessor_cff.py`.

Tested in 10_0_0_pre1, no changes expected (except possibly from the `photonPostProcessor_cff.py` fix).

@Dr15Jones

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

File: Validation/EcalDigis/python/ecalSelectiveReadoutValidation_cfi.py [hunk 1]
```diff
@@ -84,6 +84,7 @@
 
 
 from Configuration.Eras.Modifier_fastSim_cff import fastSim
-if fastSim.isChosen():
-    ecalSelectiveReadoutValidation.EbSimHitCollection = cms.InputTag("famosSimHits","EcalHitsEB")
-    ecalSelectiveReadoutValidation.EeSimHitCollection = cms.InputTag("famosSimHits","EcalHitsEE")
+fastSim.toModify(ecalSelectiveReadoutValidation,
+    EbSimHitCollection = "famosSimHits:EcalHitsEB",
+    EeSimHitCollection = "famosSimHits;EcalHitsEE"
```
Comment (perrotta) [inline Validation/EcalDigis/python/ecalSelectiveReadoutValidation_cfi.py]: Typo: ";" -> ":"

**Comment by perrotta** [comment]
@makortel : I compared the configuration dumps with and without this PR, and I see no changes, except the two typos in photonPostProcessor_cff.py that you are fixing here, and the additional typo inadvertently added by you as reported in https://github.com/cms-sw/cmssw/pull/21389#pullrequestreview-78948406 : once you fix that one, I think that this PR can be quickly signed.

**Comment by makortel** [comment]
Fixed the typo pointed by @perrotta in https://github.com/cms-sw/cmssw/pull/21389#pullrequestreview-78948406 (thanks!). I guess the PR tests not failing means that the file (`Validation/EcalDigis/python/ecalSelectiveReadoutValidation_cfi.py`) is not used in practice in the FastSim workflows (at least in those of PR tests).

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by perrotta** [comment]
IB is also broken: let wait for the next one....

**Comment by perrotta** [comment]
@cmsbuild please test

**Comment by perrotta** [comment]
please test
(Last CMSSW_10_0_X_2017-11-28-2300 IB should be fixed now)

**Comment by perrotta** [comment]
+1
- Comparing the configuration dumps with and without this PR, I see no changes, except the two typos in photonPostProcessor_cff.py which are fixed here
- No changes in jenkins tests, as expected

**Comment by davidlange6** [comment]
merge

**Pull request merged**